R/track_rust_source.R

```diff
@@ -200,3 +184,32 @@ get_file_info <- function(path) {
   info <- dplyr::mutate(info, path = rownames(info), .before = dplyr::everything())
   tibble::as_tibble(info)
 }
+
+# Find files newer than the reference file.
+#
+# @param files File paths to find newer ones than `reference`.
+# @param reference A file path to compare `files` against.
+# @return File paths newer than `reference`.
+find_newer_files_than <- function(files, reference) {
+  # If `files` is of length 0 or NULL, exit early.
+  if (length(files) == 0L) {
+    return(character(0))
+  }
+
+  if (length(reference) != 1L) {
+    stop("`reference` must be length 1 of character vector", call. = FALSE)
```

Comment:
Same with `stop` here. We could provide super-detailed error messages now.
```R
details <- character(0)
if (length(reference) != 1L) {
  details <- ui_x("Expected vector of length `1`, got `{length(reference)}`.")
}
if(typeof(reference) != "character") {
  details <- c(details, ui_x("Expected type `character`, got `{typeof(reference)}`."))
}
ui_throw("Invalid argument `reference`.", details = details)
```

Reply:
Thanks, I was wondering how detailed the error here should be, as this is more like an assertion that should be always met internally, not an user-facing error. No matter however friendly the error message is, it's probably not what user can fix.